v4/export/config.go

```diff
@@ -28,28 +33,85 @@ func DefaultConfig() *Config {
 		Logger:        &DummyLogger{},
 		FileSize:      UnspecifiedSize,
 		OutputDirPath: ".",
+		ServerInfo:    UnknownServerInfo,
+		SortByPk:      false,
 	}
 }
 
 func (conf *Config) getDSN(db string) string {
 	return fmt.Sprintf("%s:%s@tcp(%s:%d)/%s", conf.User, conf.Password, conf.Host, conf.Port, db)
 }
 
-func extractOutputConfig(conf *Config) *Config {
-	return &Config{
-		Logger:        conf.Logger,
-		FileSize:      conf.FileSize,
-		OutputDirPath: conf.OutputDirPath,
+const UnspecifiedSize = 0
+
+type ServerInfo struct {
+	ServerType    ServerType
+	ServerVersion *ServerVersion
+}
+
+var UnknownServerInfo = ServerInfo{
+	ServerType:    UnknownServerType,
+	ServerVersion: nil,
+}
+
+var versionRegex = regexp.MustCompile("^(\\d+\\.){2}\\d+")
+var tidbVersionRegex = regexp.MustCompile("v(\\d+\\.){2}\\d+")
+
+func ParseServerInfo(versionStr string) (ServerInfo, error) {
+	lowerCase := strings.ToLower(versionStr)
+	serverInfo := ServerInfo{}
+	if strings.Contains(lowerCase, "tidb") {
+		serverInfo.ServerType = TiDBServerType
+	} else if strings.Contains(lowerCase, "mariadb") {
+		serverInfo.ServerType = MariaDBServerType
+	} else if versionRegex.MatchString(lowerCase) {
+		serverInfo.ServerType = MySQLServerType
+	} else {
+		serverInfo.ServerType = UnknownServerType
+	}
+
+	var trimmedVersionStr string
+	if serverInfo.ServerType == TiDBServerType {
+		trimmedVersionStr = tidbVersionRegex.FindString(versionStr)[1:]
+	} else {
+		trimmedVersionStr = versionRegex.FindString(versionStr)
 	}
+	versionNums := strings.Split(trimmedVersionStr, ".")
+	if len(versionNums) != 3 {
+		return serverInfo, nil
+	}
+	var vs [3]int
+	var err error
+	for i, s := range versionNums {
+		vs[i], err = strconv.Atoi(s)
+		if err != nil {
+			return serverInfo, err
+		}
+	}
+
+	serverInfo.ServerVersion = makeServerVersion(vs[0], vs[1], vs[2])
+	return serverInfo, nil
 }
 
-type WriteConfig struct {
-	// Logger is used to log the export routine.
-	Logger Logger
-	// Output size limit in bytes.
-	OutputSize int
-	// OutputDirPath is the directory to output.
-	OutputDirPath string
+type ServerType int8
+
+const (
+	UnknownServerType = iota
```

Comment:
```suggestion
	UnknownServerType ServerType = iota
```

BTW better name the constants `ServerTypeUnknown` etc for better auto-completion behavior.